### PR TITLE
DCS-844 Do not retrieve latest booking after update

### PR DESCRIPTION
**Pull request by andrewrlee**
Read replica may not be up to date by the time we re-request the booking so instead reconstitute from local data

### CODE REVIEW

File: backend/services/bookingService.ts [hunk 1]
```diff
@@ -101,18 +117,38 @@ export = class BookingService {
...
       await this.whereaboutsApi.updateVideoLinkBooking(context, videoBookingId, {
         comment: update.comment,
-        ...(update.preLocation
-          ? { pre: this.toAppointment(update.preLocation, preAppointmentTimes(update.startTime)) }
-          : {}),
-        main: this.toAppointment(update.mainLocation, [update.startTime, update.endTime]),
-        ...(update.postLocation
-          ? { post: this.toAppointment(update.postLocation, postAppointmentTime(update.endTime)) }
-          : {}),
+        ...(pre ? { pre: this.toAppointment(pre) } : {}),
```
Comment (bell-pa) [inline backend/services/bookingService.ts:135]: This is a bit.. mind bending. Had to think it through quite carefully before it made sense! Could this be simpler? I'm pretty sure that pre and post can be null.  Does the whereabouts-api end object?

File: backend/services/bookingService.ts [hunk 2]
```diff
@@ -101,18 +117,38 @@ export = class BookingService {
     )
 
     if (status === 'AVAILABLE') {
+      const existing = await this.get(context, videoBookingId)
+      const locations = await this.prisonApi.getLocationsForAppointments(context, existing.agencyId)
```
Comment (bell-pa) [inline backend/services/bookingService.ts:121]: could lines 121 .. 131 be extracted to a method that returned pre,main, post?

File: backend/services/bookingService.ts [hunk 3]
```diff
@@ -101,18 +117,38 @@ export = class BookingService {
...
-        main: this.toAppointment(update.mainLocation, [update.startTime, update.endTime]),
-        ...(update.postLocation
-          ? { post: this.toAppointment(update.postLocation, postAppointmentTime(update.endTime)) }
-          : {}),
+        ...(pre ? { pre: this.toAppointment(pre) } : {}),
+        main: this.toAppointment(main),
+        ...(post ? { post: this.toAppointment(post) } : {}),
+      })
+
+      await this.notificationService.sendBookingUpdateEmails(context, currentUsername, {
```
Comment (bell-pa) [inline backend/services/bookingService.ts:140]: looks like the descriptions are only used here.  Perhaps they could be created separately, then you wouldn't need to create appointmentDto and convert o appointment. toDescriptions(locations, update) or similar.

**Review by bell-pa — APPROVED** [review]
LGTM, but with comments/suggestions for simplifying bookingService code.

**Comment by andrewrlee** [comment]
Thanks, will upgrade in a follow up

**Pull request merged**